FuelLabs/fuel-indexer
# fix: downgrade `fuel-core-client` to `v0.17.2`

### Description

This PR downgrades `fuel-core-client` to `v0.17.2` as the move to the latest version negatively impacted our ability to index information from the beta-3 testnet. It also downgrades a few other dependencies as well; special care has been taken to minimize the amount of downgrading necessary.

When either beta-3 is upgraded or a new testnet is released, we should be able to revert this PR to get rid of the manual type conversions and make only a few necessary adjustments to move to the required version.

### Testing steps

CI should pass.

#### Manual Testing

Please compile the `fuel-explorer` example and run the following command: 
```
cargo run --bin fuel-indexer -- run --manifest examples/fuel-explorer/fuel-explorer/fuel_explorer.manifest.yaml --run-migrations --verbose --fuel-node-host beta-3.fuel.network --fuel-node-port 80
```

You should see that beta-3 is being successfully indexed once more.

### Notes

For the testing suite, I had to downgrade `fuels-rs` to `v0.40` (our previously used version); this was because of difficulties with the built-in `fuel-core` helpers (for starting the test node) and working with the downgraded types.
